public/scripts/history.js

```diff
@@ -14,3 +14,30 @@ async function getHistory(questionNo) {
         return [];
     }
 }
+
+// record the time taken to answer a given question based on the question
+// number
+function recordAnswer(questionNo, timeSpentMS) {
+    if (!loadHistory()) {
+        console.error('Unable to record new time for question ' + questionNo);
+        return false;
+    }
+
+    // check for existing timing data, initialize if none found
+    if (!_history[questionNo]) {
+        _history[questionNo] = [];
+    }
+    _history[questionNo].push(timeSpentMS);
+    return true;
+}
+
+module.exports = {
+    getHistory,
+    getRemoteHistory,
+    saveRemoteHistory,
+    loadHistory,
+    saveHistory,
+    recordAnswer,
+    clearHistory
+
+  };
```

Comment:
From other commits, `recordAnswer` and `getHistory` are left so the others have to be removed from exports.